mislabel 406 on Input validation failed. Details: {invalid time range for the query, Given [3769 days] max allowed: [90 days]}"}

2023-04-28 10:07:09 [INFO] - workloader version 9.12.2 - started mislabel
2023-04-28 10:07:09 [INFO] - using default pce - xyz - 22.5.10-95
2023-04-28 10:07:10 [DEBUG] - GetTrafficAnalysis http request: POST https://scpX.illum.io:443/api/v2/orgs/1/traffic_flows/async_queries
2023-04-28 10:07:10 [DEBUG] - GetTrafficAnalysis request body: {"query_name":"","sources":{"include":[],"exclude":[]},"destinations":{"include":[],"exclude":[]},"services":{"include":[],"exclude":[{"port":5355,"proto":17},{"port":137,"proto":17},{"port":138,"proto":17},{"port":139,"proto":17}]},"start_date":"2013-01-01T00:00:00Z","end_date":"2023-04-28T10:07:09.7024313+10:00","policy_decisions":["allowed","potentially_blocked","blocked"],"max_results":100000,"exclude_workloads_from_ip_list_query":true}
2023-04-28 10:07:10 [INFO] - GetTrafficAnalysis status sode: 406
2023-04-28 10:07:10 [DEBUG] - GetTrafficAnalysis response body: [{"token":"input_validation_error","message":"Input validation failed. Details: {invalid time range for the query, Given [3769 days] max allowed: [90 days]}"}]
2023-04-28 10:07:10 [ERROR] - error making traffic api call - http status code of 406

--- a/cmd/mislabel/mislabel.go
+++ b/cmd/mislabel/mislabel.go
@@ -130,8 +130,9 @@ func misLabel() {
 	}
 
 	// Build the traffic query struct
+	// Use 88-day window (PCE max is 90 days) instead of hardcoded 2013 start date
 	tq := illumioapi.TrafficQuery{
-		StartTime:                       time.Date(2013, 1, 1, 0, 0, 0, 0, time.UTC),
+		StartTime:                       time.Now().AddDate(0, 0, -88).In(time.UTC),
 		EndTime:                         time.Now(),
 		PolicyStatuses:                  []string{"allowed", "potentially_blocked", "blocked"},
 		PortProtoExclude:                exclPorts,

